app/scripts/controllers/user-storage/encryption/encryption.ts

```diff
@@ -55,56 +66,68 @@ class EncryptorDecryptor {
   }
 
   #encryptStringV1(plaintext: string, password: string): string {
-    const salt = randomBytes(this.#PBKDF2_SALT_SIZE);
-
-    // Derive a key using PBKDF2.
-    const key = pbkdf2(sha256, password, salt, {
-      c: this.#PBKDF2_ITERATIONS,
+    const key = this.#getOrGenerateScryptKey(password, {
+      N: this.#SCRYPT_N,
+      r: this.#SCRYPT_r,
+      p: this.#SCRYPT_p,
       dkLen: this.#ALGORITHM_KEY_SIZE,
     });
 
     // Encrypt and prepend salt.
     const plaintextRaw = utf8ToBytes(plaintext);
     const ciphertextAndNonceAndSalt = concatBytes(
-      salt,
-      this.#encrypt(plaintextRaw, key),
+      key.salt,
+      this.#encrypt(plaintextRaw, key.key),
     );
 
     // Convert to Base64
     const encryptedData = byteArrayToBase64(ciphertextAndNonceAndSalt);
 
     const encryptedPayload: EncryptedPayload = {
       v: '1',
+      t: 'scrypt',
       d: encryptedData,
-      iterations: this.#PBKDF2_ITERATIONS,
+      o: {
+        N: this.#SCRYPT_N,
+        r: this.#SCRYPT_r,
+        p: this.#SCRYPT_p,
+        dkLen: this.#ALGORITHM_KEY_SIZE,
+      },
+      saltLen: this.#SCRYPT_SALT_SIZE,
     };
 
     return JSON.stringify(encryptedPayload);
   }
 
   #decryptStringV1(data: EncryptedPayload, password: string): string {
-    const { iterations, d: base64CiphertextAndNonceAndSalt } = data;
+    const { o, d: base64CiphertextAndNonceAndSalt, saltLen } = data;
 
     // Decode the base64.
     const ciphertextAndNonceAndSalt = base64ToByteArray(
       base64CiphertextAndNonceAndSalt,
     );
 
     // Create buffers of salt and ciphertextAndNonce.
-    const salt = ciphertextAndNonceAndSalt.slice(0, this.#PBKDF2_SALT_SIZE);
+    const salt = ciphertextAndNonceAndSalt.slice(0, saltLen);
     const ciphertextAndNonce = ciphertextAndNonceAndSalt.slice(
-      this.#PBKDF2_SALT_SIZE,
+      saltLen,
       ciphertextAndNonceAndSalt.length,
     );
 
-    // Derive the key using PBKDF2.
-    const key = pbkdf2(sha256, password, salt, {
-      c: iterations,
-      dkLen: this.#ALGORITHM_KEY_SIZE,
-    });
+    // Derive the key.
+    const key = this.#getOrGenerateScryptKey(
```

Comment:
(nitpick) `key` can be destructured here since the returned salt value isn't used.

Reply:
Yep good call out! On it!